javatests/arcs/core/storage/StorageProxyTest.kt

```diff
@@ -158,19 +159,55 @@ class StorageProxyTest {
         storageProxy.addOnSync("testHandle") {
             syncDeferred.complete(true)
         }
+        joinAll()
```

Comment:
I'm curious how this works. What is joinAll() waiting on here? Isn't there a chance this will finish and fall through to complete(false) before the callback fires?

Reply:
You're right, this is no good. Let me figure out a better way to do this.

Reply:
you could use suspendCoroutine { continuation -> block } trick I use in the AndroidEntityHandleManagerTest